### PR TITLE
deployer: Use dhcp on crowbar_register only when enable_pxe is set (bsc#1132654)

### PR DESCRIPTION
**Pull request by bhdn**

### CODE REVIEW

**Comment by susecloudcommitpolice** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please shorten commit subjects to less than 70 chars</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: chef/cookbooks/provisioner/templates/suse/crowbar_register.erb [hunk 1]
```diff
@@ -55,6 +60,7 @@ while test $# -gt 0; do
         <% end -%>
         --gpg-auto-import-keys) CROWBAR_AUTO_IMPORT_KEYS=1 ;;
         --no-gpg-checks) CROWBAR_NO_GPG_CHECKS=1 ;;
+        --no-dhcp) USE_DHCP=0 ;;
```
Comment (vuntz) [inline chef/cookbooks/provisioner/templates/suse/crowbar_register.erb]: I probably wouldn't add the option (not really clear when this would be useful).

Comment (bhdn) [reply chef/cookbooks/provisioner/templates/suse/crowbar_register.erb]: Indeed, it was added to match the other options only. Removed it now.

**Comment by toabctl** [comment]
@bhdn I guess we need backports for SOC8 and SOC7?

**Pull request merged**